tools/mo/openvino/tools/mo/moc_frontend/serialize.py

```diff
@@ -4,14 +4,53 @@
 import argparse
 import os
 
-from openvino.runtime import Model  # pylint: disable=no-name-in-module,import-error
+from openvino.runtime import Model, PartialShape  # pylint: disable=no-name-in-module,import-error
 
 from openvino.tools.mo.back.preprocessing import apply_preprocessing
 from openvino.tools.mo.utils.cli_parser import parse_transform
 
 
 def moc_emit_ir(ngraph_function: Model, argv: argparse.Namespace):
     output_dir = argv.output_dir if argv.output_dir != '.' else os.getcwd()
+    if argv.framework == "pytorch":
+        import torch
+        from openvino.frontend.pytorch.decoder import pt_to_ov_type_map
+        example_inputs = getattr(argv, "example_input", None)
+        input_signature = getattr(argv, "input_signature", None)
+        provide_shapes = argv.input_shape is not None
+        if example_inputs is not None:
+            inputs = [example_inputs] if isinstance(example_inputs, torch.Tensor) else example_inputs
+            if input_signature is not None and isinstance(inputs, dict):
+                ordered_inputs = []
+                upd_sign = []
+                for key in input_signature:
+                    if key not in inputs:
+                        continue
+                    ordered_inputs.append(inputs[key])
+                    upd_sign.append(key)
+                inputs = ordered_inputs
+                input_signature = upd_sign
+            for idx, input_tensor in enumerate(ngraph_function.inputs):
+                if isinstance(inputs, (list, tuple)):
+                    input_data = inputs[idx]
+                else:
+                    input_data = list(inputs.values())[idx]
+                pt_dtype = input_data.dtype if isinstance(input_data, torch.Tensor) else type(input_data)
+                dtype = pt_to_ov_type_map.get(str(pt_dtype))
+                if dtype is None:
+                    raise f"Unknown input dtype {pt_dtype}"
+
+                input_tensor.get_node().set_element_type(dtype)
+                if input_signature is not None:
+                    tensor = input_tensor.get_tensor()
+                    input_names = tensor.names
+                    input_names.update(input_signature[idx])
+                    tensor.set_names(input_names)
+                if not provide_shapes:
+                    # prevent dynamic rank issue
+                    shape = [-1] * len(input_data.shape)
+                input_tensor.get_node().set_partial_shape(PartialShape(shape))
+            ngraph_function.validate_nodes_and_infer_types() 
```

Comment:
Please do not add this code here. we have already common part for FE configuration. Already left comment about it.
Also this routine is responsible for executing MOC and pre-processing transformation and IR serialization

Reply:
I'm not familar with mo code, could you please provide link, where the right place?
For Inow,  can only guess which part of the code you mean... Also it can not be done as part of preprocessing on input model, only after convert

Reply:
I have already put the link above. Duplicate one more time: https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/openvino/tools/mo/moc_frontend/pipeline.py#L24

Reply:
ok, moved, thanks for clarification